### ISSUE TITLE
[$30] - [Bug Hunt - NEW BUG]Web page flash when no items choosen

### ISSUE THREAD

**Issue by twicoder** [topcoder-platform/community-app#176]
1) Unselect all types of contests (design, develepment, data science)
2) It's expected to see no item, but we see many empty items
3) And the web page is flash again and again after several seconds
<img width="1643" alt="2017-08-24 9 19 17" src="https://user-images.githubusercontent.com/3046795/29668372-85030ffc-8912-11e7-9541-38b29a05da66.png">

**Comment by birdofpreyru** [comment]
Accepted, 3 points.
### Expected Fix
Indeed, at the moment the frontend code does not know that such (or, probably, some others as well) combination of filters cannot satisfy any challenge in principle. So, the frontend just keeps trying to find past challenges matching these fitlers. We need some elegant solution to avoid it.

**Comment by ordinaryname** [comment]
Can you assign me to this issue please?

**Comment by birdofpreyru** [comment]
@ordinaryname Added you to collaborators. Can you check, whether READ access is enough to self-assign?

**Comment by ordinaryname** [comment]
I cannot self-assign

**Comment by birdofpreyru** [comment]
@ordinaryname and now?

**Comment by ordinaryname** [comment]
Fixed issue with Pull Request #323

**Comment by birdofpreyru** [comment]
A simple and a bit hacky solution has been merged in by myself.
Though, with this solution our frontend still does one cycle of search of past challenges when all track filters are unset. We are still interested in a better fix (i.e. when we set a combination of filters which cannot match anything at all, we should not start the regular search procedure for past challenges). And it should not focus to match on the tracks switches, in a sense that it should be straightforward to add other checks of filters state, when this workaround should be used.
